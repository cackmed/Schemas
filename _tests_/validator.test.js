const Validator  = require('../lib/validator.js');

describe('Validator', () => {
  let validator;

  describe('required fields', () => {
    beforeAll(() => {
      validator = new Validator('age', {
        type: Number,
        required: true
      });
    });

    it('returns the field', () => {
      const person = {
        name: 'Joe',
        age: 23,
        feathers: false
      };

      expect(validator.validate(person)).toEqual(23);
    });

    it('returns the field cast to type', () => {
      const person = {
        name: 'Joe',
        age: 23,
        feathers: false
      };

      expect(validator.validate(person)).toEqual(23);
    });

  });
});
