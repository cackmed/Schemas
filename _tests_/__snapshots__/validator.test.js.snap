// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator required fields returns the field 1`] = `"Cannot read property 'validate' of undefined"`;
